### DISCUSSION TITLE
@click events don't trigger in x-html rendered nested elements

### DISCUSSION THREAD

**Discussion by dgigafox** [alpinejs/alpine#1603] [7. Bugs (archived)]
## Summary
Elements with a click event that are dynamically rendered using `x-html` isn't triggred

## Steps to reproduce
I prepared 2 codepens to reproduce it:
v2 - https://codepen.io/dgigafox/pen/QWpXNyZ
v3 - https://codepen.io/dgigafox/pen/abJgZPK

Here is the html layout that both codepen are using:
```html
<div x-data="data()" id="root">
    <template x-for="color in colors">
      <div x-html="content(color)"></div>
    </template>
  
  <button @click="activeColor = 'No color'">No color</button>
  
  <div>Color is: <span x-text="activeColor"></span></div>
</div>

<script>
  function data() {
    return {
  colors: ['Red', 'Orange', 'Yellow'], activeColor: null,
  setColor(color) { this.activeColor = color },
  content(color) { return `<button @click="setColor('${color}')" >${color}</button>` }
}
  }
</script>

```

## Actual results
In v2, clicking on the Red, Orange, Yellow buttons changes the `activeColor` which is displayed at the bottom. Those buttons are dynamically rendered via `x-html` directive. Although in v3, with the same layout, click events aren't triggered - only the `No color` button is triggered since it is already present from the start.

## Expected result
The corresponding color of the button should be set when the click event is triggered just like in v2 and should as work as well in v3

**Comment by KevinBatdorf** [comment]
There's a solution presented here you can try for now: https://github.com/alpinejs/alpine/discussions/1594#discussioncomment-906534
